bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/BookieJournalForceTest.java

```diff
@@ -0,0 +1,405 @@
+        Journal journal = new Journal(journalDir, conf, ledgerDirsManager);
+
+        CountDownLatch forceWriteThreadSuspendedIsWaiting = new CountDownLatch(1);
+        LinkedBlockingQueue<ForceWriteRequest> supportQueue = new LinkedBlockingQueue<>();
+        @SuppressWarnings("unchecked")
+        BlockingQueue<ForceWriteRequest> forceWriteRequests = mock(BlockingQueue.class);
+        AtomicBoolean forceWriteThreadSuspended = new AtomicBoolean(true);
+
+        doAnswer((Answer) (InvocationOnMock iom) -> {
+            supportQueue.put(iom.getArgument(0));
+            return null;
+        }).when(forceWriteRequests).put(any(ForceWriteRequest.class));
+
+        when(forceWriteRequests.take()).thenAnswer(new Answer<ForceWriteRequest>() {
+            @Override
+            public ForceWriteRequest answer(InvocationOnMock iom) throws Throwable {
+                forceWriteThreadSuspendedIsWaiting.countDown();
+                while (forceWriteThreadSuspended.get()) {
+                    Thread.sleep(100);
+                }
+                ForceWriteRequest res =  supportQueue.take();
+                while (forceWriteThreadSuspended.get()) {
+                    Thread.sleep(100);
+                }
+                return res;
+            }
+        });
+        Whitebox.setInternalState(journal, "forceWriteRequests", forceWriteRequests);
+        journal.start();
+        assertTrue(forceWriteThreadSuspendedIsWaiting.await(10, TimeUnit.SECONDS));
+
+        CountDownLatch latch = new CountDownLatch(1);
+        AtomicInteger countForceWriteCallsBeforeAck = new AtomicInteger();
+        replace(method(JournalChannel.class, "forceWrite")).with(new InvocationHandler() {
+            @Override
+            public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+                log.info("{} called with arguments {} ", method.getName(), Arrays.toString(args));
+                if (latch.getCount() == 1) {
+                    countForceWriteCallsBeforeAck.incrementAndGet();
+                }
+                try {
+                    method.invoke(proxy, args);
+                } catch (InvocationTargetException err) {
+                    throw err.getCause();
+                }
+                return null;
+            }
+        });
+
+        // wait for first file to be created
+        while (countForceWriteCallsBeforeAck.get() == 0) {
+            Thread.sleep(100);
+        }
+
+        long ledgerId = 1;
+        long entryId = 0;
+        journal.logAddEntry(ledgerId, entryId, DATA, true /* ackBeforeSync */, new WriteCallback() {
+            @Override
+            public void writeComplete(int rc, long ledgerId, long entryId, BookieSocketAddress addr, Object ctx) {
+                latch.countDown();
+            }
+        }, null);
+        assertTrue(latch.await(20, TimeUnit.SECONDS));
+        assertTrue(supportQueue.isEmpty());
```

Comment:
better assert that `forceWrite` is not called.

Reply:
done